onebusaway-android/src/main/java/org/onebusaway/android/io/ObaContext.java

```diff
@@ -93,36 +93,8 @@ public void setBaseUrl(Context context, Uri.Builder builder) {
         String serverName = Application.get().getCustomApiUrl();
 
         if (!TextUtils.isEmpty(serverName) || mRegion != null) {
-            Uri baseUrl = null;
-            if (!TextUtils.isEmpty(serverName)) {
-                Log.d(TAG, "Using custom API URL set by user '" + serverName + "'.");
-
-                try {
-                    // URI.parse() doesn't tell us if the scheme is missing, so use URL() instead (#126)
-                    URL url = new URL(serverName);
-                } catch (MalformedURLException e) {
-                    // Assume HTTP scheme, since without a scheme the Uri won't parse the authority
-                    serverName = context.getString(R.string.http_prefix) + serverName;
-                }
-
-                baseUrl = Uri.parse(serverName);
-            } else if (mRegion != null) {
-                Log.d(TAG, "Using region base URL '" + mRegion.getObaBaseUrl() + "'.");
-
-                baseUrl = Uri.parse(mRegion.getObaBaseUrl());
-            }
-
-            // Copy partial path (if one exists) from the base URL
-            Uri.Builder path = new Uri.Builder();
-            path.encodedPath(baseUrl.getEncodedPath());
-
-            // Then, tack on the rest of the REST API method path from the Uri.Builder that was passed in
-            path.appendEncodedPath(builder.build().getPath());
-
-            // Finally, overwrite builder that was passed in with the full URL
-            builder.scheme(baseUrl.getScheme());
-            builder.encodedAuthority(baseUrl.getEncodedAuthority());
-            builder.encodedPath(path.build().getEncodedPath());
+            Log.d(TAG, "Using custom OBA API URL set by user '" + serverName + "'.");
```

Comment:
This log statement should be moved to within `setUrl()` inside of the ` if (!TextUtils.isEmpty(serverName)) {` IF clause, so it's only logged if the user actually set a custom OBA API URL via Preferences.